LanguageFeatures/Patterns/type_inference_A24_t03.dart

```diff
@@ -0,0 +1,85 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion
+/// To type check a pattern p being matched against a value of type M:
+/// ...
+/// Variable:
+///
+/// If the variable has a type annotation, the required type of p is that type,
+/// as is the static type of the variable introduced by p.
+///
+/// Else the required type of p is M, as is the static type of the variable
+/// introduced by p
+///
+/// @description Check that if a variable pattern has no type annotation, then
```

Comment:
```suggestion
/// @description Check that if a variable pattern `p` has no type annotation, then
```

We know that `p` must be that variable pattern, but it's better style to introduce every name which is used.